File: VB-Examples/10_Shapes/ShapeToImage.vb
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet.Shapes
Imports System.Drawing.Imaging

Namespace ShapeToImage
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load an existing Excel file into the workbook
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ShapeToImage.xlsx")

            ' Get the first worksheet from the workbook
            Dim sheet1 As Worksheet = workbook.Worksheets(0)

            ' Get the first PrstGeomShape in the worksheet and cast it to XlsShape type
            Dim shape As XlsShape = TryCast(sheet1.PrstGeomShapes(0), XlsShape)

            ' Save the shape as an image
            Dim img As Image = shape.SaveToImage()

            ' Save the image to a file named "ShapeToImage.png" in PNG format
            img.Save("ShapeToImage.png", ImageFormat.Png)
            ' Release the resources used by the workbook
            workbook.Dispose()
            FileViewer("ShapeToImage.png")
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/10_Shapes/ShapeToImage.cs
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet.Shapes;
using System;
using System.Drawing;
using System.Drawing.Imaging;
using System.Windows.Forms;

namespace ShapeToImage
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load the Excel document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ShapeToImage.xlsx");

            //Get the first worksheet
            Worksheet sheet1 = workbook.Worksheets[0];

            //Get the first shape from the first worksheet
            XlsShape shape = sheet1.PrstGeomShapes[0] as XlsShape;

            //Save the shape to a image
            Image img = shape.SaveToImage();
            img.Save("ShapeToImage.png", ImageFormat.Png);


            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            FileViewer("ShapeToImage.png");
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
